src/operator/operator_common.h

```diff
@@ -314,6 +314,32 @@ inline bool dispatch_mode_assign(DispatchMode *y, const DispatchMode& x) {
   }
 #endif
 
+/*! \brief allocate ndarrays from existing ndarrays
```

Comment:
Not sure if this is the best place to put this function. Would moving this inside custom.h work? If we put it here it's very likely that somebody misuses the fucntion looking at the current doc